src/twisted/names/dns.py

```diff
@@ -378,7 +379,36 @@ def _isSubdomainOf(descendantName, ancestorName):
     return descendantLabels[-len(ancestorLabels) :] == ancestorLabels
 
 
-def str2time(s):
+def _str2time(s: str) -> int:
```

Comment:
Maybe add a comment with the reason why we need  this separate/helper method.

I see you left a comment in the PR... but I think that any comment that you fell is needed to be added to the PR, is best to have it as a code comment.